src/components/PlaylistModal/PlaylistModal.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import "./PlaylistModal.scss";
+import { IoCloseCircle, IoAddOutline } from "react-icons/io5";
+import { addVideoToPlaylist } from "../../utilities/Playlist/add-playlist-video";
+import { useVideo } from "../../context/VideoContext";
+import { addPlaylist } from "../../utilities/Playlist/add-playlist";
+import { removeVideoFromPlaylist } from "../../utilities/Playlist/remove-playlist-video";
+
+const PlaylistModal = ({ setShowPlaylistModal, videoDetails }) => {
+    const {
+        stateVideo: { playlists },
+        dispatchVideo,
+    } = useVideo();
+
+    const [showInput, setShowInput] = useState(false);
+    const [playlistTitle, setPlaylistTitle] = useState("");
+
+    const inPlaylist = (videoId, playlistId, playlists) =>
+        playlists
+            ?.find((playlist) => playlist._id === playlistId)
+            ?.videos.some((playlist) => playlist._id === videoId);
```

Comment:
rather than playlist it is better to name it video

Reply:
Thanks, I forgot to change it since I just needed to check if the given id is present. Will do it now.